xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/XWiki.java

```
@@ -1516,25 +1516,17 @@ public InputStream getResourceAsStream(String s) throws MalformedURLException
 
     public String getResourceContent(String name) throws IOException
     {
-        InputStream is = null;
         if (getEngineContext() != null) {
-
-            try {
-                is = getResourceAsStream(name);
+            try (InputStream is = getResourceAsStream(name)) {
+                if (is == null) {
+                    // Resources should always be encoded as UTF-8, to reduce the dependency on the system encoding
+                    return FileUtils.readFileToString(new File(name), DEFAULT_ENCODING);
+                }
+                return IOUtils.toString(is, DEFAULT_ENCODING);
             } catch (Exception e) {
```

There is no reason to catch all exceptions like this.